pkg/storage/pebble_mvcc_scanner.go

```diff
@@ -1521,6 +1521,12 @@ func (p *pebbleMVCCScanner) processRangeKeys(seeked bool, reverse bool) bool {
 		if hasPoint {
 			return true
 		}
+
+		// Stepping might move the iterator entirely off the bare range key, at
+		// which point the previous curRangeKeys buffers are invalidated.
+		p.curRangeKeys.CloneInto(&p.savedRangeKeys)
+		p.curRangeKeys = p.savedRangeKeys
```

Comment:
This would only happen for an exhausted iterator, since we'd otherwise loop around and detect this via `RangeKeyChanged()`. Why are we keeping the range keys around in this case? I think the right fix would be to set `p.curRangeKeys = MVCCRangeKeyStack{}` in the `if !p.iterValid()` branch below.

Reply:
> Why are we keeping the range keys around in this case?

In tombstone mode we still care that there was a MVCC range tombstone, even if there's no point.

Reply:
Oh, for `get()`, right. The implementation there is a bit hacky, there's probably a better way to do it. I think we should try to avoid doing this unnecessary copying in the general case for scans. Would be happy to have a look at it when I have a chance, if this isn't urgent.

Reply:
Making copies only in tombstone mode seems ok, we don't use that very often. I'll see if I can improve this separately.

Reply:
Submitted an alternative in #97327 which I think is slightly cleaner and only pays the cost in the one specific case that we need it (a tombstone point get).

Reply:
Nevermind, #97327 won't work, we'll have to do something like what you're doing here. We can consider adding an `isGet` field (I think we used to have one?) and only enable it in that case, but there aren't currently any callers that set `MVCCScanOptions.Tombstones` so it isn't particularly important right now.